gulpfile.ts

```diff
@@ -0,0 +1,226 @@
+import gulp from 'gulp';
+import tap from 'gulp-tap';
+import filter from 'gulp-filter';
+import merge from 'merge-stream';
+import ts from 'gulp-typescript';
+import terser from 'gulp-terser';
+import GulpUglify from 'gulp-uglify';
+import eslint from 'gulp-eslint-new';
+import babel from 'gulp-babel';
+import { SrcOptions } from 'vinyl-fs';
+import sourcemaps from 'gulp-sourcemaps';
+import tsMacroTransformer, { macros } from 'ts-macros';
+import pathsTransformer from 'ts-transform-paths';
+import rollupTypescript from 'rollup-plugin-typescript2';
+import { CustomTransformerFactory, Program } from 'typescript';
+import minifyPrivatesTransformer from 'ts-transformer-minify-privates';
+
+/* eslint-disable @typescript-eslint/no-var-requires */
+const clean = require('gulp-clean') as () => NodeJS.ReadWriteStream;
+const rollup = require('gulp-rollup-2').rollup as (...params: unknown[]) => NodeJS.ReadWriteStream;
```

Comment:
Why convert the type here?

Reply:
Because they don't have a type definition, they become any type.

There is a problem that any propagate during pipe chaining, so I gave a proper type.